test/ruby/test_zjit.rb

```diff
@@ -1011,6 +975,15 @@ def test(x)
     }, insns: [:opt_duparray_send], call_threshold: 1
   end
 
+  def test_opt_newarray_send_hash
+    assert_compiles 'Integer', %q{
+      def test(x)
+        [1, 2, x].hash
+      end
+      test(20).class
+    }, insns: [:opt_newarray_send], call_threshold: 1
```

Comment:
let's also add a case where `Array#hash` is overridden

Reply:
Done in 796d59c2d88dabc6cd0b0fabd4875d56e83c4aa5. Please verify that's what you had in mind.